### ISSUE TITLE
[DISCUSS] `Aptamer` and `experiments` misnomer

### ISSUE THREAD

**Issue by satvshr** [gc-os-ai/pyaptamer#101]
* The [Aptamer  class](https://github.com/gc-os-ai/pyaptamer/blob/6f0bea7c7a166456f840417b359f1708e840c08f/pyaptamer/experiments/_aptamer.py#L10) seems to be a misnomer (for me) as it sounds like it provides functionalities similar to [`Seq`](https://biopython.org/docs/latest/Tutorial/chapter_seq_objects.html) from `biopython`, whereas it is a class for candidate evaluation. I suggest renaming it to `NNeval`(given it takes a `torch.nn.Module` object) or something similar.
* Given what Franz said that `all "experiments" would implement evaluators` we should rename the `experiments` folder to something more recognisable at first glance, like `evaluators`.

**Comment by fkiraly** [comment]
PS: why the "invalid" label? I think this is typically used by maintainers to mark an issue as "invalid" (e.g., can be ignored)

**Comment by satvshr** [comment]
> PS: why the "invalid" label? I think this is typically used by maintainers to mark an issue as "invalid" (e.g., can be ignored)

Oh, I just put the tag based on the description "it does not seem right".....which it didn't.

**Comment by fkiraly** [comment]
> Why not benchmarks?

That might work, but it is a bit overloaded. A benchmark is usually a collection of experiments with fixed parameter settings.
(FYI @jgyasu, @benheid, if you are watching)

**Comment by satvshr** [comment]
@fkiraly `evaluation` and `analysis` are 2 good suggestions GPT could come up with.

> That might work, but it is a bit overloaded. A benchmark is usually a collection of experiments with fixed parameter settings.

`benchamarks` imply what we are going for though, maybe to fit the definition, we create a sub-directory inside `benchmarks` called `evaluators`? So someone loking for evaluators will first look at `benchmarks `since they are closely related terms, and then `evaluators`?

**Comment by NennoMP** [comment]
> > seems to be a misnomer (for me) as it sounds like it provides functionalities similar to [Seq](https://biopython.org/docs/latest/Tutorial/chapter_seq_objects.html) from biopython, whereas it is a class for candidate evaluation
> 
> I agree with this. Let's find a better name. [@NennoMP](https://github.com/NennoMP)

@fkiraly Oh yeah, I agree. I discussed this topic with Satvik a little on discord.

Personally, I am fine with calling the folder `experiments` as in `hyperactive`. Mostly because I don't think we have better options? `evals`, `benchmarks`, etc. do not feel that much better but with `experiments` we are at least "consistent" with  `hyperactive`.

That said, `Aptamer` could be renamed to something like `AptamerEval`, in the sense that we are evaluating an aptamer candidate and giving it a score?

**Comment by satvshr** [comment]
> That said, `Aptamer` could be renamed to something like `AptamerEval`, in the sense that we are evaluating an aptamer candidate and giving it a score?

`AptamerNNEval `maybe? Given we might expand on it later into a framework that can accept `sklearn `compatible models too?

**Comment by fkiraly** [comment]
> Hmm, but it still seems like a small link not big enough to try naming directories in both similarly.

I see it growing in the future - since optimization algorithms seem to be important, in particular as variants of in-silico selex.

**Comment by satvshr** [comment]
> > Hmm, but it still seems like a small link not big enough to try naming directories in both similarly.
> 
> I see it growing in the future - since optimization algorithms seem to be important, in particular as variants of in-silico selex.

Oh, fair point then, but then again I do not think it is clear for someone new to the library, after understanding the purpose behind the directory it only got more confusing for me 😅
